W-15608556 updating expiring certificates jh (#35)
mulesoft/docs-api-experience-hub: modules/ROOT/pages/updating-expiring-certificates.adoc

= Updating Expiring Certificates
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> Connected apps enable a two-way connection between Anypoint Platform and Salesforce organizations. The connected apps are dependent on digital certificates for proper functionality. These certificates are time-bound and subject to expiration. 

Reviewer: ```suggestion
Connected apps enable a two-way connection between Anypoint Platform and Salesforce organizations. The connected apps depend on digital certificates for proper functionality. These certificates are time-bound and subject to expiration. 
```

== How Certificates Are Generated

> API Experience Hub generates a few self-signed certificates during the connection process that expire in one year.

Reviewer: ```suggestion
During the connection process, API Experience Hub generates self-signed certificates that expire in one year.
```

> A self-signed certificate is generated to link the Anypoint Platform organization and the Salesforce organization. The certificate and the private key is stored in Secrets Manager in Anypoint Security. When this certificate expires, no notification is sent to the administer of the Salesforce instance. However, the initial connection between Anypoint Platform and Salesforce remains operational. The next time the administrator logs in to API Experience Hub, an error message displays a connection error and enables the administrator to reconnect. When the administrator logs back in to API Experience Hub, a new certificate generates  automatically and expires in one year. 

Reviewer: ```suggestion
One of these self-signed certificates links the Anypoint Platform organization and the Salesforce organization. The certificate and the private key are stored in Secrets Manager in Anypoint Security. When this certificate expires, no notification is sent to the administrator of the Salesforce instance. However, the initial connection between Anypoint Platform and Salesforce remains operational. The next time the administrator logs in to API Experience Hub, an error message displays a connection error and enables the administrator to reconnect. When the administrator logs back in to API Experience Hub, a new certificate generates automatically with a one-year expiration time. 
```

> API Experience Hub uses the Salesforce Metadata API to generate an Anypoint certificate during the connection process. The connected app uses this certificate to access metadata content from Anypoint. This certificate is also valid for one year. The administrator of the Salesforce instance receives an email 30 days before the certificate expires. Because a named credential uses the Anypoint certificate, the certificate cannot be deleted initially. To generate a new Anypoint certificate, rename the certificate to trigger a diagnostic check to fail and create a new Anypoint certificate. After renaming the certificate, the portal administrator can log back in to API Experience Hub successfully. As a best practice, delete the renamed certificate because the maximum number of Salesforce certificates is 50.

Reviewer: ```suggestion
API Experience Hub uses the Salesforce Metadata API to generate an Anypoint certificate during the connection process. The connected app uses this certificate to access metadata content from Anypoint. This certificate is also valid for one year. The administrator of the Salesforce instance receives an email 30 days before the certificate expires. Because a named credential uses the Anypoint certificate, you cannot delete the certificate. To generate a new Anypoint certificate, rename the existing certificate to cause a diagnostic check to fail, then create a new Anypoint certificate. After renaming the certificate, the portal administrator can log back in to API Experience Hub successfully. Ensure you delete the renamed certificate because the maximum number of Salesforce certificates is 50.
```
Maybe you can separate this into two paragraphs when the instructions start: _"To generate a new Anypoint certificate..."_

Update: On further reading, it seems the steps here are the same that are explained in detail in the task sub-section. Does it make sense to explain this two times? Or are they different things? Maybe you can just omit everything after _To generate a new Anypoint certificate..._
Please check what seems better in this context.

Author: Thanks, this makes sense. I will remove the redundant info.

== Before You Begin
Before getting started, ensure you have the API Experience Hub Administrator permissions.

== Update the Salesforce Identity Data Certificate 
To generate a new Anypoint certificate:

. Go to Salesforce and click *Setup* > *Security* > *Certificate and Key Management*.
. Locate *Anypoint* in the list and click *Edit*.
. Enter a new name, for example, *Anypoint to expire*. 
+
image::aeh-rename-cert.png[Example for renaming the certificate]
. Click *Save*.

> . Log in to API Experience Hub to trigger a diagnostic check to create a new certificate.

Reviewer: Does the diagnostic check generate the new certificate? From the previous steps, it seems the user needs to create a new certificate after this check fails. Please check if this is correct.

Author: Yes, the check generates a new certificate.
